app/routes/datasets/datasets.py

```diff
@@ -16,9 +17,17 @@
     tags=["Datasets"],
     response_model=DatasetsResponse,
 )
-async def get_datasets() -> DatasetsResponse:
+async def get_datasets(params: Params = Depends()) -> DatasetsResponse:
     """Get list of all datasets."""
+    if params.size < 50:
+        query = datasets.get_datasets_query()
+        data = await paginate(query, params)
+        adapted = [dataset.__values__ for dataset in data.items]
+        return DatasetsResponse(data=adapted,
+                                    links={'self': "", 'first': "", 'last': "", 'prev': "", 'next': ""},
+                                    meta={'size': data.size, 'total-pages': "", 'total-items': data.total},
+                                    status='success')
 
     data = await datasets.get_datasets()
+    return DatasetsResponse(data=data, status='success')
```

Comment:
This actually turned out to be the hardest part for introducing a breaking changes _without_ actually breaking things. The main idea was to have the same endpoint _do the right thing_ if pagination params were given but otherwise remain unchanged to preserve legacy behavior.

With `pydantic` I couldn't just create subclasses of `DatasetsResponse` and continue to return the generalized one. When I did actually tried to do that, `pydantic` would throw a ValidationError and tell me that I was either including extra fields or it would add the pagination properties when they should've been hidden.